lib/src/e2ee/key_provider.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:typed_data';
+
+import 'package:flutter_webrtc/flutter_webrtc.dart' as rtc;
+
+const defaultRatchetSalt = 'LKFrameEncryptionKey';
+const defualtMagicBytes = 'LK-ROCKS';
```

Comment:
nit: typo

Reply:
also: maybe it makes sense for the lk-server to tell us what magic bytes it's using?
1. we wouldn't have to hard-code it across clients
2. server implementations could change them over time

cc @boks1971

Reply:
Yeah, that would be good. It can be random per session/track too. Where should be notify it? In `TrackInfo`?

Reply:
`TrackPublishedResponse` maybe, not sure. I guess making it configurable per track would mean some additional wrangling on the client side. Any particular scenarios you have in mind for setting it per track rather than session wide?

Reply:
No reason for track wide. Session wide should be fine. Just going with the principles of as granular/as flexible as possible.

Not sure about `TrackPublishedResponse`. The injection happens in down track path. So, subscribe only clients also will need it.

Also, my `TrackInfo` suggestion is bad because it is a publisher side stuff.

Maybe, a separate message for this?